kotlin/group-layers/src/main/java/com/esri/arcgisruntime/sample/grouplayers/LayerListAdapter.kt

```diff
@@ -0,0 +1,172 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.sample.grouplayers
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.CheckBox
+import android.widget.LinearLayout
+import android.widget.RadioButton
+import android.widget.RadioGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.layers.GroupLayer
+import com.esri.arcgisruntime.layers.GroupVisibilityMode
+import com.esri.arcgisruntime.layers.Layer
+import com.esri.arcgisruntime.mapping.LayerList
+
+/**
+ * A custom RecyclerView.Adapter to display group layers and sublayers, accounting for group layers
+ * with an exclusive visibility mode.
+ *
+ * @param dataSet the list of layers for the scene
+ * @param onLayerCheckedChanged a callback function which is invoked by each layer and sublayer's onCheckedChangedListener
+ */
+class LayerListAdapter(
+  private val dataSet: LayerList,
+  private val onLayerCheckedChanged: (layer: Layer, isChecked: Boolean) -> Unit
+) :
+  RecyclerView.Adapter<LayerListAdapter.ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+    // There are two view holder types, for independent and exclusive visibility modes
+    return when (viewType) {
+      TYPE_EXCLUSIVE -> {
+        val v = LayoutInflater.from(parent.context).inflate(
+          R.layout.radio_group, parent, false
+        )
+        ExclusiveLayerViewHolder(v) as ViewHolder
+      }
+      else -> {
+        val v = LayoutInflater.from(parent.context).inflate(
+          R.layout.checkbox_group, parent, false
+        )
+        DefaultLayerViewHolder(v) as ViewHolder
+      }
+    }
+  }
+
+
+  override fun getItemViewType(position: Int): Int {
+    val layer = dataSet[position]
+    return if (layer is GroupLayer && layer.visibilityMode == GroupVisibilityMode.EXCLUSIVE) {
+      TYPE_EXCLUSIVE
+    } else TYPE_DEFAULT
+  }
+
+  override fun getItemCount(): Int = dataSet.size
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      TYPE_EXCLUSIVE -> (holder as ExclusiveLayerViewHolder).let { exclusiveLayerViewHolder ->
+        val layer = dataSet[position]
+        exclusiveLayerViewHolder.layer = layer
+        exclusiveLayerViewHolder.textView.text = layer.name
+        exclusiveLayerViewHolder.onLayerChecked = onLayerCheckedChanged
+        exclusiveLayerViewHolder.sublayers.apply {
+          clear()
+          addAll((layer as GroupLayer).layers)
+        }
+        exclusiveLayerViewHolder.populate()
```

Comment:
```suggestion
        val childLayer = dataSet[position]
        exclusiveLayerViewHolder.apply {
          layer = childLayer
          textView.text = childLayer.name
          onLayerChecked = onLayerCheckedChanged
          sublayers.clear() 
          sublayers.addAll((childLayer as GroupLayer).layers)
          populate()
        }
```

Reply:
Might be a good chance to use an apply block?

Am I correct is renaming layer -> childLayer?

Up to you if you want to make these changes--if so, could replicate it on the defaultLayerViewHolder

Reply:
this one should not be a child layer, because in theory it's either a group layer or a standalone layer not contained in a group layer

Reply:
as for the apply block, I'm happy to make that change--I was only trying to avoid too much nesting